temboardagent/plugins/maintenance/functions.py

```diff
@@ -435,3 +445,161 @@ def get_table(conn, schema, table):
+
+    if res.type == taskmanager.MSG_TYPE_ERROR:
+        logger.error(res.content)
+        raise HTTPError(500, "Unable to schedule vacuum")
+
+    return res.content
+
+
+def vacuum(conn, dbname, schema, table, mode):
+    # Run vacuum statement
+    # Check that the specified table exists in schema
+    conn.execute(
+        "SELECT 1 FROM pg_tables WHERE tablename = '{table}' AND "
+        "schemaname = '{schema}'".format(table=table, schema=schema)
+    )
+    if not list(conn.get_rows()):
+        raise UserError("Table %s.%s not found" % (schema, table))
+
+    # Build the SQL query
+    q = "VACUUM"
+    if mode == 'full':
+        q += " FULL"
+    elif mode == 'analyze':
+        q += " ANALYZE"
+    elif mode == 'freeze':
+        q += " FREEZE"
+    q += " {schema}.{table}".format(schema=schema, table=table)
+
+    try:
+        # Try to execute the statement
+        logger.info("Running SQL on DB %s: %s" % (dbname, q))
+        conn.execute(q)
+        logger.info("VACCUM done.")
+    except error as e:
+        logger.exception(str(e))
+        logger.error("Unable to execute SQL: %s" % q)
+        raise UserError("Unable to run vacuum %s on %s.%s"
+                        % (mode, schema, table,))
+
+
+def task_status_label(status):
+    labels=['todo', 'scheduled', 'queued', 'doing', 'done', 'failed',
+            'canceled', 'aborted', 'abort']
+    p = status.bit_length() - 1
+    try:
+        return labels[p]
+    except IndexError:
+        return 'unknown'
+
+
+def list_scheduled_vacuum(app):
+    # Get list of scheduled vacuum operations
+    ret = []
+    try:
+        # Ask it to the task manager
+        tasks = taskmanager.TaskManager.send_message(
+            str(os.path.join(app.config.temboard.home, '.tm.socket')),
+            taskmanager.Message(taskmanager.MSG_TYPE_TASK_LIST, ''),
+            authkey=None,
+        )
+    except Exception as e:
+        logger.exception(str(e))
+        raise HTTPError(500, "Unable to get scheduled vacuum list")
+
+    for task in tasks:
+        # We only want vacuum tasks
+        if task['worker_name'] != 'vacuum_worker':
+            continue
+        ret.append(dict(
+            id=task['id'],
+            dbname=task['options'].get('dbname'),
+            schema=task['options'].get('schema'),
+            table=task['options'].get('table'),
+            mode=task['options'].get('mode'),
+            datetime=task['start_datetime'].strftime("%Y-%m-%dT%H:%M:%SZ"),
+            status=task_status_label(task['status'])
+        ))
+    return ret
+
+
+def cancel_scheduled_vacuum(id, app):
+    # Cancel one scheduled vacuum operation. If the vacuum is running, the task
+    # is going to be aborted.
+
+    # Check the id
+    f = False
+    for vacuum_task in list_scheduled_vacuum(app):
+        if vacuum_task['id'] == id:
+            f = True
+            break
+    if not f:
+        raise HTTPError(404, "Scheduled vacuum operation not found")
```

Comment:
if not id in [t['id'] for t in list_scheduled_vacuum(app)]:
     raise HTTPError...